Goal: Don't try to contact the CUPS server when we can reliably
determine that no printers are needed

Fixes: bug #479512

Upstream status: submitted as bugzilla bug #5525

Index: samba-3.0.30/source/smbd/server.c
===================================================================
--- samba-3.0.30/source/smbd/server.c	(revision 1918)
+++ samba-3.0.30/source/smbd/server.c	(working copy)
@@ -624,6 +624,10 @@
 	int pnum = lp_servicenumber(PRINTERS_NAME);
 	const char *pname;
 
+	if (!lp_load_printers()
+	    && (lp_auto_services() == NULL || !strcmp(lp_auto_services(),"")))
+		return;
+
 	pcap_cache_reload();
 
 	/* remove stale printers */
